dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/PreCheckValidateAndGenerateUidHook.java

```diff
@@ -120,7 +121,27 @@ public void validateEvent( ValidationErrorReporter reporter, Event event )
         validateNotesUid( event.getNotes(), reporter );
     }
 
-    private void validateNotesUid( List<Note> notes, ValidationErrorReporter reporter )
+    @Override
+    public void validateRelationship( ValidationErrorReporter reporter, Relationship relationship )
```

Comment:
In my refactor, I am actually generating all missing uids. The only thing we need to validate is that isUidInvalid(). This is just a note though, for this PR, keep it like this :)